src/ext/java/org/opentripplanner/ext/vehicleparking/hslpark/HslHubToVehicleParkingGroupMapper.java

```diff
@@ -0,0 +1,94 @@
+package org.opentripplanner.ext.vehicleparking.hslpark;
+
+import com.bedatadriven.jackson.datatype.jts.parsers.GenericGeometryParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.locationtech.jts.geom.Geometry;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingGroup;
+import org.opentripplanner.transit.model.basic.I18NString;
+import org.opentripplanner.transit.model.basic.NonLocalizedString;
+import org.opentripplanner.transit.model.basic.TranslatedString;
+import org.opentripplanner.transit.model.basic.WgsCoordinate;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.util.geometry.GeometryUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Maps a HSL Park hub into a {@link VehicleParkingGroup}.
+ */
+public class HslHubToVehicleParkingGroupMapper {
+
+  private static final Logger log = LoggerFactory.getLogger(
+    HslHubToVehicleParkingGroupMapper.class
+  );
+
+  private static final GenericGeometryParser GEOMETRY_PARSER = new GenericGeometryParser(
+    GeometryUtils.getGeometryFactory()
+  );
+
+  private final String feedId;
+
+  public HslHubToVehicleParkingGroupMapper(String feedId) {
+    this.feedId = feedId;
+  }
+
+  public Map<FeedScopedId, VehicleParkingGroup> parseHub(JsonNode jsonNode) {
+    var hubId = HslParkToVehicleParkingMapper.createIdForNode(jsonNode, "id", feedId);
+    try {
+      Map<String, String> translations = new HashMap<>();
+      JsonNode nameNode = jsonNode.path("name");
+      nameNode
+        .fieldNames()
+        .forEachRemaining(lang -> {
+          String name = nameNode.path(lang).asText();
+          if (!name.equals("")) {
+            translations.put(lang, nameNode.path(lang).asText());
+          }
+        });
+      I18NString name = translations.isEmpty()
+        ? new NonLocalizedString(hubId.getId())
+        : TranslatedString.getI18NString(translations, true, false);
+      Geometry geometry = GEOMETRY_PARSER.geometryFromJson(jsonNode.path("location"));
+      double x = geometry.getCentroid().getX();
```

Comment:
We could add a new constructor for `WgsCoordinate`, which we could use here (and in couple of other places)